Prepare for CI server

gregoryjscott opened the pull request:
This sets up Rake tasks for the CI server (and for developers). This simplifies what the CI server has to run and allows us to change what the CI server does by simply changing the rake task and pushing to GitHub. The CI server will run the following commands after fetching the latest version of the repository from GitHub.

- `bundle install` - This will install/update any gems that have been added to the Gemfile.
- `rake ci:staging` - This will run the CI task for the environment (currently just performs a Jekyll build with the appropriate configuration). The CI job for production will call `rake ci:prod` instead.

Additional changes:
- `_build.yml` is replaced by `_config.prod.yml` file that only contains the values that override and/or append to what is in `_config.yml`. This will fix #29.
- The Rake tasks take care of passing the correct configuration for each environment.
- The Rake tasks take care of running `bundle exec`.
- The default Rake task is set to run Jekyll build locally using `staging` values. This means that developers just need to type `rake` at the command line instead of `bundle exec jekyll build`.

Tasks
- [x] Create separate tasks for staging and production

gregoryjscott commented:
This is ready to go. I created separate Rake tasks for staging and production and got rid of the redundant configuration. I updated the description of this PR with more details. @bechatman please review.

bechatman commented:
Makes sense to me, from the little that I understand about how multiple config files are served. This is showing a merge conflict on the site, though, so I am guessing merging that update to Jekyll did something.

gregoryjscott commented:
@bechatman re: the configuration, I found this jekyll issue helpful in understanding how it works https://github.com/jekyll/jekyll/issues/958. The Rake tasks are doing the exact thing described in that issue.

I rebased this and fixed the conflict. I'm going to merge it so we'll have it ready for setting up the CI server this afternoon.
